src/jit/optimizer.cpp

```diff
@@ -6203,6 +6203,12 @@ bool Compiler::optHoistLoopExprsForTree(
         {
             treeIsHoistable = false;
         }
+
+        // Avoid to hoist reference loading
+        if (tree->OperGet() == GT_CLS_VAR && tree->gtType == TYP_BYREF)
+        {
+            treeIsHoistable = treeIsInvariant = false;
+        }
```

Comment:
Also our coding convention do not permit multiple assignments in one statement:

Do NOT:  ` treeIsHoistable = treeIsInvariant = false;`

Instead:
```
 treeIsHoistable = false;
 treeIsInvariant = false;
```